.github/workflows/test-platforms.yml

```diff
@@ -7,21 +7,63 @@ on:
     types: [opened, synchronize, reopened]
 
 jobs:
-  Build:
+  job:
+    name: ${{ matrix.os }} - ${{ github.workflow }}
     runs-on: ${{ matrix.os }}
     strategy:
+      fail-fast: false
       matrix:
         os: [ubuntu-latest, macos-latest, windows-latest]
+        include:
+          - os: windows-latest
+            triplet: x64-windows
+          - os: ubuntu-latest
+            triplet: x64-linux
+          - os: macos-latest
+            triplet: x64-osx
+
+    env:
+      VCPKG_DEFAULT_TRIPLET: ${{ matrix.triplet }}
+
     steps:
       - uses: actions/checkout@v2
-      # build project
-      - name: make build directory
+        with:
+          submodules: true
+
+      - name: Vcpkg bootstrap Windows
+        run: bootstrap-vcpkg.bat
+        working-directory: vcpkg
+        if: ${{ matrix.os == 'windows-latest' }}
+
+      - name: Vcpkg bootstrap Shell
+        run: ./bootstrap-vcpkg.sh
+        shell: bash
+        working-directory: vcpkg
+        if: ${{ matrix.os != 'windows-latest' }}
+
+      - name: Mac Install Brew
+        run: brew install automake libtool
```

Comment:
You might need to include "pkg-config" here

Reply:
I'm not familiar with that. I will check it out.